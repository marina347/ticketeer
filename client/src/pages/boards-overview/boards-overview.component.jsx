import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import WelcomeTeaser from "../../components/welcome-teaser/welcome-teaser.component";
import BoardItemList from "../../components/board-item-list/board-item-list.container";
import AddBoard from "../../components/add-board-item/add-board-item";
import { BoardsOverviewPageContainer } from "./boards-overview.styles";
import { selectUserNameAndSurname } from "../../redux/user/user.selectors";

const BoardsOverviewPage = ({ name }) => {
  return (
    <BoardsOverviewPageContainer className="nice-font">
      <WelcomeTeaser displayName={name} />
      <AddBoard />
      <BoardItemList />
    </BoardsOverviewPageContainer>
  );
};

export const mapStateToProps = createStructuredSelector({
  name: selectUserNameAndSurname,
});

export default connect(mapStateToProps)(BoardsOverviewPage);
