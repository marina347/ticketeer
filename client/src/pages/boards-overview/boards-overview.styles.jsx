import styled from "styled-components";

export const BoardsOverviewPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
