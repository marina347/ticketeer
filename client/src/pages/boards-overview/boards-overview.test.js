import React from "react";
import { shallow } from "enzyme";
import { BoardsOverviewPage } from "./boards-overview.component";

describe("Boards overview page", () => {
  let wrapper;
  let name = "John";

  beforeEach(() => {
    wrapper = shallow(<BoardsOverviewPage name={name} />);
  });

  it("Should render Board overview page", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
