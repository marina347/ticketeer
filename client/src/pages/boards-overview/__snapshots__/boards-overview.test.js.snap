// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boards overview page Should render Board overview page 1`] = `
<div
  className="boards-overview"
>
  <WelcomeTeaser
    displayName="John"
  />
  <WrappedComponent />
  <withRouter(Connect(Connect(BoardItemList))) />
</div>
`;
